### PR TITLE
Check if the address is valid when using hexdump

### LINKED ISSUE #2832: Hexdump doesn't show an adress is unmapped
This seems like a bug, although it doesn't error out, huh.

```
pwndbg> hexdump 0xdeadbeef
+0000 0xdeadbeef

pwndbg> vmmap 0xdeadbeef
There are no mappings for specified address or module.
```

Pwndbg version:
```
pwndbg> version
Pwndbg:   2025.02.19 build: 97ea040b
Python:   3.11.6 (main, Apr 10 2024, 17:26:07) [GCC 13.2.0]
GDB:      14.0.50.20230907-git
Capstone: 5.0.3
Unicorn:  2.1.0
Pwnlib:   4.14.0
```

### PR DESCRIPTION
**Pull request by dkvhr**
I used the same approach telescope uses to check if the address is valid by peeking at it.
There is also a change at the test that checks an empty hexdump

Closes #2832

### CODE REVIEW

**Review by patryk4815 — APPROVED** [review]

File: pwndbg/commands/hexdump.py [hunk 1]
```diff
@@ -77,6 +77,9 @@ def hexdump(address, count=pwndbg.config.hexdump_bytes) -> None:
         hexdump.offset = 0
 
     address = int(address)
+    if not pwndbg.aglib.memory.peek(address):
```
Comment (disconnect3d) [inline pwndbg/commands/hexdump.py:80]: Shouldn't we move this after the next if condition?

Comment (disconnect3d) [reply pwndbg/commands/hexdump.py:80]: Pls send another PR moving this further if u don't mind

Comment (dkvhr) [reply pwndbg/commands/hexdump.py:80]: sure! I will send another one. Thanks for the review :)
